scripts/workarounds/boot-order/metal-lib.sh

```diff
@@ -0,0 +1,510 @@
+[ -z "$efibootmgr_prefix" ] && export efibootmgr_prefix=''
+
+function efi_fail_host {
+    echo >&2 "no prefix-driver for hostname: $hostname"
+    return 1
+}
+
+function efi_trim {
+    echo disabling undesired boot entries $(cat /tmp/rbbs*) && cat /tmp/rbbs* | sort | sed 's/^Boot//g' | awk '{print $1}' | tr -d '*' | xargs -r -i efibootmgr -b {} -A
+}
+
+function efi_remove {
+    echo removing undesired boot entries $(cat /tmp/sbbs*) && cat /tmp/sbbs* | sort | sed 's/^Boot//g' | awk '{print $1}' | tr -d '*' | xargs -r -i efibootmgr -b {} -B
+}
+
+function efi_enforce {
+    # IMPORTANT: The ENTIRE list of entries needs to exist, otherwise iLO/HPE servers will undo any changes.
+    # both /tmp/bbs* and /tmp/rbbs* are concatenated together; the ordinal order of the /tmp/bbsNUM files
+    # will enforce NICs first.
+    local boot_order_bbs
+    local boot_order_rbbs
+    local boot_order
+    boot_order_bbs=$(cat /tmp/bbs*   | sed 's/^Boot//g' | awk '{print $1} ' | tr -d '*' | tr -d '\n' | sed -r 's/(.{4})/\1,/g;s/,$//')
+    boot_order_rbbs=$(cat /tmp/rbbs* | sed 's/^Boot//g' | awk '{print $1} ' | tr -d '*' | tr -d '\n' | sed -r 's/(.{4})/\1,/g;s/,$//')
+    # Note: if $efibootmgr_prefix is set, it will already contain the necessary trailing comma
+    # remove trailing commas in case 'cat /tmp/rbbs*' is empty
+    boot_order=$(echo ${efibootmgr_prefix}${boot_order_bbs},${boot_order_rbbs} | sed 's/,*$//g')
+    echo enforcing boot order $(cat /tmp/bbs*) && efibootmgr -o $boot_order | grep -i bootorder
+    echo activating boot entries && cat /tmp/bbs* | awk '!x[$0]++' | sed 's/^Boot//g' | tr -d '*' | awk '{print $1}' | xargs -r -i efibootmgr -b {} -a
+}
+
+# uses /tmp/rbbs99
+function efi_specials {
+    # TODO: If Marvell; then ensure PXE retries only once per NIC.
+    echo 'removing Shasta V1.3 items' && efibootmgr | grep -iP '(crayinstall|sles-secureboot)' | tee /tmp/sbbs
+}
+
+function setup_uefi_bootorder() {
+cat << EOM
+Configuring UEFI boot-order...
+these use the same commands from the manual page:
+    https://github.com/Cray-HPE/docs-csm/blob/main/background/ncn_boot_workflow.md#setting-order
+EOM
+    echo scanning vendor ... && vendor=$(ipmitool fru | grep -i 'board mfg' | tail -n 1 | cut -d ':' -f2 | tr -d ' ')
+    hostname=${hostname:-$(hostname)}
+    # Add vendors here; add like-vendors on the same case statement.
+    # "like-vendors" means their efibootmgr outboot matches
+
+    # formatting:
+    # if another vendor is identical then it should live with another.
+    # vendors may have differing hostnames, depending where this script runs
+    # vendor)
+    #   hostname_prefix_1)
+    #     file1)
+    #     fileN)
+    #   hostname_prefix_2)
+    #     file1)
+    #     fileN)
+    #   hostname_prefix_N)
+    #     file1)
+    #     fileN)
+    #   error)
+    #   remove_file_1
+    #   remove_file_N
+    # done
+    case $vendor in
+        *GIGABYTE*)
+            # Removal file(s) ...
```

Comment:
There are multiple removal files here.
```suggestion
            # Removal files ...
```